### ISSUE TITLE
API Documentation

### ISSUE THREAD

**Issue by navinpai** [Slash-Go/Docs#1]
Document the SlashGo Server API in either `README` or in a more in-depth format like `OpenAPI` to allow easy reference to API methods, fields and possible edge cases.

**Comment by zahertalab** [comment]
Is it a RESTful API? If yes, where I can start?

**Comment by navinpai** [comment]
Hey @zahertalab ,

Yes, it is a restful API. The API routes are all part of the [api directory](https://github.com/Slash-Go/SlashGoServer/tree/main/api) and they call code in the [controllers directory](https://github.com/Slash-Go/SlashGoServer/tree/main/controllers) which can be used as a reference for validations etc.

**Comment by zahertalab** [comment]
I wonder if there is some swagger or open API UI available?

**Comment by navinpai** [comment]
We don't have Swagger/OpenAPI set up as of today, but if you can set up a swagger yaml, I'd be happy to have it part of the project to make it more accessible to folks in the future! :+1: 

**Update**: @zahertalab  We now have a new swagger documentation repo: https://github.com/Slash-Go/Docs ... I've moved the issue to the new repo. Docs are currently served at https://docs.slashgo.link/  from this repo.

**Comment by EleekaN** [comment]
Hi @navinpai I want to work on this. Just to summarize, you want the swagger documentation to be updated with other end points like   `Auth` , `Organizations` etc.

**Comment by navinpai** [comment]
> Hi @navinpai I want to work on this. Just to summarize, you want the swagger documentation to be updated with other end points like `Auth` , `Organizations` etc.

Yes, that's right... It would be great to have the [swagger.yaml](https://github.com/Slash-Go/Docs/blob/main/swagger.yaml) updated with all the endpoints and responses

Thanks in advance!

**Comment by EleekaN** [comment]
Sure...is there one location/file where I can get all the endpoints and
responses. I remember seeing some of it on different folders.

**Comment by navinpai** [comment]
> Sure...is there one location/file where I can get all the endpoints and responses. I remember seeing some of it on different folders.

The API routes are all part of the [api directory](https://github.com/Slash-Go/SlashGoServer/tree/main/api) and they call code in the [controllers directory](https://github.com/Slash-Go/SlashGoServer/tree/main/controllers) which can be used as a reference for validations etc. Sorry we don't have a single path for it yet, but we're hoping this repo becomes that single source of truth for the API :smile_cat:

**Comment by EleekaN** [comment]
localhost://3000 doesn't reflect the changes I make on the swagger.yaml file. How can I sync it locally, so I can validate my work?

**Comment by navinpai** [comment]
Once you make changes in swagger file, please do the following steps from [the README](https://github.com/Slash-Go/Docs/blob/main/README.md) to update the generated `index.html` file with your changes:

```
// If you don't have redoc-cli, install it
$ npm i -g redoc-cli serve

$ redoc-cli build swagger.yaml -o index.html
$ serve -s .
```

**Comment by EleekaN** [comment]
Hi @navinpai, I have done some baseline work for 'auth' endpoints. Need some inputs and feedback before I proceed further . I will create a pull request, please have  a look.

**Comment by Peaaboo-Madton** [comment]
Hello, do you still need help with the documentation? I see the last update was 28 days ago.

**Comment by akltech** [comment]
checking in if you still need help with the documentation
